### PR TITLE
VZ-8314 tools analysis e2e test for InsufficientCPU case

### PR DESCRIPTION
**Pull request by absBitm**

### CODE REVIEW

File: tools/vz/pkg/analysis/internal/util/report/issue.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ const (
 	ImagePullNotFound         = "ImagePullNotFound"
 	ImagePullService          = "ImagePullService"
 	InsufficientMemory        = "InsufficientMemory"
-	InsufficientCPU           = "InsufficientCpu"
+	InsufficientCPU           = "InsufficientCPU"
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/report/issue.go:104]: Is this change made to match the runbook file content/en/docs/troubleshooting/diagnostictools/analysisadvice/InsufficientCPU.md ?

File: tests/e2e/verify-analyze-tool/node/node_test.go [hunk 2]
```diff
@@ -44,15 +49,25 @@ var _ = t.Describe("VZ Tools", Label("f:vz-tools-node-issues"), func() {
...
 			Eventually(func() bool {
 				return utility.VerifyIssue(utility.ReportAnalysis[utility.InsufficientMemory][1], utility.InsufficientMemory)
 			}, waitTimeout, pollingInterval).Should(BeFalse())
 		})
+		t.It("Should Have InsufficientCPU Issue Post Bad CPU Request", func() {
+			Eventually(func() bool {
+				return utility.VerifyIssue(utility.ReportAnalysis[utility.InsufficientCPU][0], utility.InsufficientCPU)
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+		})
+		t.It("Should Not Have InsufficientCPU Issue Post Rectifying CPU Request", func() {
```
Comment (lake-of-dreams) [inline tests/e2e/verify-analyze-tool/node/node_test.go:67]: Maybe I don't understand it correctly, but what call rectified the issue? from where is `patch` being called in-between ?

File: tests/e2e/verify-analyze-tool/node/node_test.go [hunk 3]
```diff
@@ -44,15 +49,25 @@ var _ = t.Describe("VZ Tools", Label("f:vz-tools-node-issues"), func() {
...
+			Eventually(func() bool {
+				return utility.VerifyIssue(utility.ReportAnalysis[utility.InsufficientCPU][1], utility.InsufficientCPU)
```
Comment (lake-of-dreams) [inline tests/e2e/verify-analyze-tool/node/node_test.go:69]: why is `utility.VerifyIssue` which is just a string comparison being called inside an `Eventually` ?

**Review by bhatpmk — APPROVED** [review]
LGTM

File: tools/vz/pkg/analysis/internal/util/report/issue.go (hunk 1 above)
Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/report/issue.go:104]: No this is not related to .md files cleanup changes

File: tests/e2e/verify-analyze-tool/node/node_test.go (hunk 2 above)
Comment (absBitm) [reply tests/e2e/verify-analyze-tool/node/node_test.go:67]: It first calls the patch method which patches the deployment with a bad injection to spawn the issue and feeds its analysis report to utility.ReportAnalysis[type_of_issue][zeroth_index]
followed by it again patches the deployment with good injection to revive the issue and feeds it analysis report to utility.ReportAnalysis[type_of_issue][first_index]

then rest of the ginkgo specifications runs and verifies the report being analysed

File: tests/e2e/verify-analyze-tool/node/node_test.go (hunk 3 above)
Comment (absBitm) [reply tests/e2e/verify-analyze-tool/node/node_test.go:69]: It first calls the patch method which patches the deployment with a bad injection to spawn the issue and feeds its analysis report to utility.ReportAnalysis[type_of_issue][zeroth_index]
followed by it again patches the deployment with good injection to revive the issue and feeds it analysis report to utility.ReportAnalysis[type_of_issue][first_index]

Post Feeding the reports, rest of the specifications verifies the report under utility.ReportAnalysis[type_of_issue], It verifies that at 0th index report (utility.ReportAnalysis[type_of_issue][zeroth_index]) must contain the type_of_issue and at 1st index report (utility.ReportAnalysis[type_of_issue][first_index]) must not contain the type_of_issue

**Pull request merged**